lib/src/Base/Geom/Mesh.cxx

```diff
@@ -65,14 +65,17 @@ Mesh::Mesh(const Sample & vertices)
 
 /* Parameters constructor, simplified interface for 1D case */
 Mesh::Mesh(const Sample & vertices,
-           const IndicesCollection & simplices)
+           const IndicesCollection & simplices,
+           const Bool checkMeshValidity)
   : PersistentObject()
   , dimension_(vertices.getDimension())
   , vertices_(0, vertices.getDimension())
   , simplices_(simplices)
 {
   // Use the vertices accessor to initialize the kd-tree
   setVertices(vertices);
+  if (checkMeshValidity)
```

Comment:
maybe we could enable it by default for small meshes if < Mesh-LargeSize

Reply:
Indeed. I had this in mind

Reply:
@jschueller something like that?

Reply:
yes

Reply:
Finally I dropped the use of several RM keys for that purpose and rely only on the specific `checkMeshValidity ` one